tracing-core/src/field.rs

```diff
@@ -428,6 +428,19 @@ where
     }
 }
 
+impl<'a, T: ?Sized> crate::sealed::Sealed for &'a mut T where T: Value + crate::sealed::Sealed + 'a {}
+
+impl<'a, T: ?Sized> Value for &'a mut T
+where
+    T: Value + 'a,
+{
+    fn record(&self, key: &Field, visitor: &mut dyn Visit) {
+        // Don't use `(*self).record(key, visitor)`, otherwise would
+        // cause stack overflow due to `unconditional_recursion`.
+        T::record(self, key, visitor)
+    }
+}
```

Comment:
I think it makes sense to add this in `tracing-core`, but would it be possible to make this change in a separate PR, and have _this_ PR only change `tracing-attributes`? That way, the Git history is a little easier to read.

Thanks!